bin/packages/watch.js

```diff
@@ -26,63 +25,55 @@ const exists = ( filename ) => {
 // Exclude test files including .js files inside of __tests__ or test folders
 // and files with a suffix of .test or .spec (e.g. blocks.test.js),
 // and deceitful source-like files, such as editor swap files.
-const isSourceFile = ( filename ) => {
-	return (
+const isSourceFile = ( filename, skip ) => {
+	const isSource =
+		/.\.(js|json|scss)$/.test( filename ) &&
+		/\/src\/.+/.test( filename ) &&
 		! [
-			/\/(benchmark|__mocks__|__tests__|test|storybook|stories)\/.+.js$/,
+			/\/(benchmark|__mocks__|__tests__|test|storybook|stories)\/.+/,
 			/.\.(spec|test)\.js$/,
-		].some( ( regex ) => regex.test( filename ) ) &&
-		/.\.(js|json|scss)$/.test( filename )
-	);
+		].some( ( regex ) => regex.test( filename ) );
+
+	return isSource ? true : skip;
 };
 
-const rebuild = ( filename ) => filesToBuild.set( filename, true );
+function getSrcDirectory( filename ) {
+	const matches = /packages\/(.+)\/src\/.+/g.exec( filename );
+	if ( ! matches || ! matches?.length ) {
+		return;
+	}
+	const [ , srcDirectory ] = matches;
+	return srcDirectory;
+}
 
-getPackages().forEach( ( p ) => {
-	const srcDir = path.resolve( p, 'src' );
-	try {
-		fs.accessSync( srcDir, fs.F_OK );
-		watch(
-			path.resolve( p, 'src' ),
-			{ recursive: true, delay: 500 },
-			( event, filename ) => {
-				if ( ! isSourceFile( filename ) ) {
-					return;
-				}
+const rebuild = ( filename ) => filesToBuild.set( filename, true );
 
-				const filePath = path.resolve( srcDir, filename );
-				if ( event === 'update' && exists( filePath ) ) {
-					console.log(
-						chalk.green( '->' ),
-						`${ event }: ${ filename }`
-					);
-					rebuild( filePath );
-				} else {
-					const buildFile = path.resolve(
-						srcDir,
-						'..',
-						'build',
-						filename
-					);
-					try {
-						fs.unlinkSync( buildFile );
-						process.stdout.write(
-							chalk.red( '  \u2022 ' ) +
-								path.relative(
-									path.resolve( srcDir, '..', '..' ),
-									buildFile
-								) +
-								' (deleted)' +
-								'\n'
-						);
-					} catch ( e ) {}
-				}
-			}
-		);
-	} catch ( e ) {
-		// doesn't exist
+watch(
+	PACKAGES_DIR,
+	{ recursive: true, delay: 500, filter: isSourceFile },
+	( event, filename ) => {
+		const filePath = path.resolve( PACKAGES_DIR, filename );
+		const srcDir = getSrcDirectory( filePath );
+		if ( event === 'update' && exists( filePath ) ) {
+			console.log( chalk.green( '->' ), `${ event }: ${ filePath }` );
+			rebuild( filePath );
+		} else {
+			const buildFile = path.resolve( srcDir, '..', 'build', filename );
+			try {
+				fs.unlinkSync( buildFile );
```

Comment:
This wasn't actually working in `trunk`. Any errors were being swallowed. The issue is fixed now.